groups/reconcile.go

```diff
@@ -311,12 +421,19 @@ func (gc *GroupsConfig) Load(rootDir string, restrictions *RestrictionsConfig) e
 	log.Printf("reading groups.yaml files recursively at %s", rootDir)
 
 	return filepath.Walk(rootDir, func(path string, info os.FileInfo, err error) error {
+		// check if the current path is accessible or not.
+		if err != nil {
+			return err
+		}
```

Comment:
This looks like new logic.  Ideally we'd see this in a separate commit from a refactor, with a test case that proves why it's needed